source/common/http/conn_manager_impl.cc

```diff
@@ -1800,9 +1817,9 @@ Tracing::OperationName ConnectionManagerImpl::ActiveStream::operationName() cons
   return connection_manager_.config_.tracingConfig()->operation_name_;
 }
 
-const std::vector<Http::LowerCaseString>&
-ConnectionManagerImpl::ActiveStream::requestHeadersForTags() const {
-  return connection_manager_.config_.tracingConfig()->request_headers_for_tags_;
+const Tracing::CustomTagMap& ConnectionManagerImpl::ActiveStream::customTags() const {
+  ASSERT(tracing_custom_tags_ != nullptr);
```

Comment:
How is this safe? Don't you have to handle case in which this has not been initialized? I think it should remain uninitialized even in the case where there is a tracing config which is quite common?

Reply:
I think it'll be called only after being initialized. It would also be better to just return a constant pointer which left it more clear, I'll fix it.